tests/integration/cache_test.go

```diff
@@ -605,9 +890,50 @@ func TestCacheUnsupportedWatchOptions(t *testing.T) {
 			if !resp.Canceled {
 				t.Errorf("expected Canceled=true, got %+v", resp)
 			}
-			if !strings.Contains(resp.Err().Error(), cache.ErrUnsupportedWatch.Error()) {
+			if !strings.Contains(resp.Err().Error(), cache.ErrUnsupportedRequest.Error()) {
 				t.Errorf("expected ErrUnsupportedWatch text %q, got %v",
-					cache.ErrUnsupportedWatch.Error(), resp.Err())
+					cache.ErrUnsupportedRequest.Error(), resp.Err())
+			}
+		})
+	}
+}
+
+func TestCacheUnsupportedGetOptions(t *testing.T) {
+	integration.BeforeTest(t)
+	clus := integration.NewCluster(t, &integration.ClusterConfig{Size: 1})
+	t.Cleanup(func() { clus.Terminate(t) })
+	client := clus.Client(0)
+
+	c, err := cache.New(client, "", cache.WithHistoryWindowSize(1))
+	if err != nil {
+		t.Fatalf("cache.New: %v", err)
+	}
+	defer c.Close()
+	if err := c.WaitReady(t.Context()); err != nil {
+		t.Fatalf("cache not ready: %v", err)
+	}
+
+	unsupported := []struct {
+		name string
+		opt  clientv3.OpOption
+	}{
+		{"WithSerializable", clientv3.WithSerializable()},
```

Comment:
Technically the cache now doesn't follow the same default as client. By default client requests are Linearizable meaning newest globally (approximate definition), which is reverse of Serializable meaning newest locally (approximate definition). While cache only serves newest locally to cache. 

Could we make that tests in cache always passes WithSerializable and it's a required setting?

Reply:
+1 to what @serathius said

I was thinking more about this earlier, and I see that the existing grpcproxy serves a response from its cache if WithSerializable() is set
https://github.com/etcd-io/etcd/blob/f79fc2c4d6d9500e175c70520b50793dd8063ffb/server/proxy/grpcproxy/kv.go#L42

For the cache here, do we want to reject requests that does _not_ have this set for now? And maybe in a future iteration, make a call into etcd?